### ISSUE TITLE
Kiosk Apps with WindowsAppSDK / WinUI3

### ISSUE THREAD

**Issue by myokeeh** [microsoft/WindowsAppSDK#3642]
I need to be able to set WindowsAppSDK / WinUI3 apps as Kiosk apps. Please work with the Windows team.

### Discussed in https://github.com/microsoft/WindowsAppSDK/discussions/2653

<div type='discussions-op-text'>

<sup>Originally posted by **myokeeh** June 22, 2022</sup>
@andrewleader, what is the latest on running WindowsAppSDK / WinUI3 apps as kiosk apps? I'm in the middle of converting a UWP kiosk app only to discover this is still unsupported. I've added my thoughts on product board some time ago and provided an update recently.</div>

**Comment by chigy** [comment]
Kiosk mode has two configurations. Single and multi-app.
•	Single-app kiosk mode – UWP is supported, Win32 is not supported.
•	Multi-app kiosk mode – both UWP and Win32 are supported.

The team who owns kiosk mode does not have an immediate plan to support Single-app so I believe the best bet is to use the multi-app mode.
[Set up a single-app kiosk on Windows - Configure Windows | Microsoft Learn](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fwindows%2Fconfiguration%2Fkiosk-single-app&data=05%7C01%7Cchigy%40microsoft.com%7Cd45253ea28fa4c955f1208db66aeb8ff%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638216672902127413%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=GRgbPXd3uM0HbKnMRycVmd3pWu0gtXU8cVlzqu4MSUU%3D&reserved=0)
[Set up a multi-app kiosk on Windows 10 - Configure Windows | Microsoft Learn](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fwindows%2Fconfiguration%2Flock-down-windows-10-to-specific-apps&data=05%7C01%7Cchigy%40microsoft.com%7Cd45253ea28fa4c955f1208db66aeb8ff%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638216672902127413%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=l%2BKWMyzWqK1fglL2caz%2FQ2clhk1Woe1rcop%2FRanlQqk%3D&reserved=0)
[Set up a multi-app kiosk on Windows 11 - Configure Windows | Microsoft Learn](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fwindows%2Fconfiguration%2Flock-down-windows-11-to-specific-apps&data=05%7C01%7Cchigy%40microsoft.com%7Cd45253ea28fa4c955f1208db66aeb8ff%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638216672902127413%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=W2vomkfaz9MF3NB3GqDJcI3zS3nGTDYsuZ2InQxKpH4%3D&reserved=0)

**Comment by myokeeh** [comment]
@chigy is this also for Store distributed WindowsAppSDK / WinUI3 apps? How can I justify having users setup kiosk mode like this when it was so easy with UWP apps? The previous steps with UWP were tolerable as it is point and click. With the steps provided in the links above, it makes the app not worthwhile dealing with. I can't stick to UWP, I can't use WindowsAppSDK / WinUI3. What a strange state to be in.

**Comment by chigy** [comment]
@myokeeh , I am waiting to hear from folks who own this feature who can answer your question.

**Comment by myokeeh** [comment]
@chigy following up on this.

**Comment by christiannagel** [comment]
Having single-app kiosk mode would be great with WinUI, but it would also be helpful to have support for WinUI packaged apps with muli-app kiosk mode. As also mentioned in https://github.com/microsoft/microsoft-ui-xaml/issues/9828 I also didn't succeed configuring kiosk mode for a packaged WinUI app.

@codendone @chigy could you reopen this issue? Discussion: #2653

**Comment by myokeeh** [comment]
@codendone @chigy yes. I would really like to get this resolved. It's still a very relevant problem for me. I keep getting pressure because of dwindling support/interest in UWP, BUT I can't get it ported over because of problems like this.

**Comment by soroshsabz** [comment]
@chigy 

I think all of UWP features must be ported to WinUI 3

**Comment by chigy** [comment]
@christiannagel , @myokeeh , @soroshsabz , just FYI, I am reaching out to the team who owns Kiosk mode but they have not yet responded back to me to comment.

**Comment by christopher-carmichael** [comment]
> [@christiannagel](https://github.com/christiannagel) , [@myokeeh](https://github.com/myokeeh) , [@soroshsabz](https://github.com/soroshsabz) , just FYI, I am reaching out to the team who owns Kiosk mode but they have not yet responded back to me to comment.

Has there been a reply yet? Personally "just use multi-app kiosk mode" is not an acceptable solution for my organization.

**Comment by brettchangus** [comment]
This is a critical issue for us as well. We have many users that ran our app in kiosk mode. Since "upgrading" from XF to MAUI, we have lost this functionality.

**Comment by JafarH96** [comment]
Unfortunately, this feature (Kiosk single-app mode) has not yet been added to WinUI3!

**Comment by myokeeh** [comment]
@chigy following up.

**Comment by eproasim** [comment]
@chigy Any updates for this?

**Comment by sjorsmiltenburg** [comment]
After migrating from Xamarin to Maui, I am also encountering this problem. Any updates?

**Comment by myokeeh** [comment]
Wow. Still no progress on this?

**Comment by mominshaikhdev** [comment]
It requires OS level changes. A new windows build.

**Comment by myokeeh** [comment]
> It requires OS level changes. A new windows build.

I believe that was already known, but now I wonder if that is planned.

**Comment by soroshsabz** [comment]
Any update?

**Comment by arknu** [comment]
If WinUI and Windows App SDK is supposed to be the premiere native development platform for modern Windows, how can there still be Windows features that don't support apps created with the native development environment for Windows? This is the problem with Microsoft, you never get around to finishing things. This should have been fixed the moment WinUi 3 and Windows App SDK was launched.
